### ISSUE TITLE
Bug Report: `vtcombo` does not respect `-mysql_server_version`

### ISSUE THREAD

**Issue by aquarapid** [vitessio/vitess#10193]
### Overview of the Issue

`vtcombo` (and possibly other components) relies on `-mysql_server_version` to setup collation code appropriately (MySQL 5.7 vs 8.0).  See this code:

https://github.com/vitessio/vitess/blob/main/go/mysql/collations/local.go#L32-L41

Unfortunately, this does not work correctly if you do not have MySQL 8.0.  This is because the `collation` module is init-ed via the module init from `planbuilder`;  and thus runs long before the CLI flags (including `-mysql_server_version`) is parsed via `flag.Parse()`

### Reproduction Steps

 * Instrument https://github.com/vitessio/vitess/blob/main/go/mysql/collations/local.go#L32-L41 with a print statement to print the value of `*servenv.MySQLServerVersion` when it is used.
 * Run vtcombo, providing `-mysql_server_version` of (for example) `5.7.30-vitess`.
 * Observe from the output of the print statement that the value of `*servenv.MySQLServerVersion` is the default value (empty string) at the time the collation code in `local.go` is run.

### Binary Version

```sh
Recent main:  31c9780fd396612c7427c57e240f8b83c8b6f3ca
```

### Operating System and Environment details

```sh
Fedora 35, x86_64.
```

### Log Fragments

```sh
Instrumenting the code in `local.go` to print a backtrace, shows the init code path that results in the `collation` module being initialized (early), before `flag.Parse()` runs:

 * https://github.com/vitessio/vitess/blob/main/go/vt/vtgate/planbuilder/system_variables.go#L30 to
 * https://github.com/vitessio/vitess/blob/main/go/vt/vtgate/planbuilder/system_variables.go#L52 to
 * https://github.com/vitessio/vitess/blob/main/go/vt/vtgate/planbuilder/system_variables.go#L65 to
 * https://github.com/vitessio/vitess/blob/main/go/vt/vtgate/evalengine/translate.go#L571 to
 * https://github.com/vitessio/vitess/blob/main/go/vt/vtgate/evalengine/translate.go#L562 to
 * https://github.com/vitessio/vitess/blob/main/go/mysql/collations/local.go#L47 to
 * https://github.com/vitessio/vitess/blob/main/go/mysql/collations/local.go#L34
```

**Comment by aquarapid** [comment]
A naive PR like https://github.com/vitessio/vitess/pull/10177 to delay the planbuilder variable init does not fix this, since we have tests (and probably other code paths) that expects these variables to be fully initialized by module load/init.

**Comment by deepthi** [comment]
@vitessio/query-serving

**Comment by vmg** [comment]
Yeah, this is bad, this basically makes the `Local()` environment always misconfigured. I was not expecting `Local()` to be called from an `init` function -- it was not designed to operate like that. I'll fix this.

**Comment by vmg** [comment]
This should now be fixed in `vtcombo` and anywhere else where we're passing a custom `-mysql_server_version` to the binary.
